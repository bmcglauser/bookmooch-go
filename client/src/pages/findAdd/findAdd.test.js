import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import FindAddPage from './';

describe('Find add page', () => {
  let page;
  beforeEach(() => {
    page = render(
      <MemoryRouter>
        <FindAddPage />
      </MemoryRouter>
    )
  })
  
  it('successfully renders the page', () => {
    expect(page).toMatchSnapshot();
  });

  it('seach should be captured', async () => {
    const input = screen.getByRole('textbox');
    fireEvent.change(input, {
      target: { value: 'norris' }
    })
    expect(input.value).toBe('norris');
  });
});