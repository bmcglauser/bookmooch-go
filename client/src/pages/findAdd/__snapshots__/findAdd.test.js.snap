// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find add page successfully renders the page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main-header"
      >
        <img
          alt="BookMooch Go logo"
          class="logo"
          src="main-logo.svg"
        />
        <h1
          class="title"
        >
          Find a book to add
        </h1>
      </div>
      <div
        class="add-page-grand-wrapper"
      >
        <div
          class="find-book-to-add"
        >
          <div
            class="top-block"
          >
            <h2>
              Enter the ISBN or ASIN code here
            </h2>
            <input
              type="text"
            />
          </div>
          <p>
            Click below to find the book you want to add. On the book's details page you'll be able to "Add this book to give away"
          </p>
          <a
            href="/searchresults//true"
            style="text-decoration: none;"
          >
            <button
              class="find-add-button"
            >
              <p>
                Find my
                <br />
                book to add
              </p>
            </button>
          </a>
        </div>
        <div
          class="add-page-footer"
        >
          <a
            href="/profile"
            style="text-decoration: none;"
          >
            <div
              class="user-button-wrapper"
            >
              <button
                class="user-button"
              />
              <p>
                User Home
              </p>
            </div>
          </a>
          <p>
            for books not yet in the catalogue, you'll have to enter the details manually on desktop
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main-header"
    >
      <img
        alt="BookMooch Go logo"
        class="logo"
        src="main-logo.svg"
      />
      <h1
        class="title"
      >
        Find a book to add
      </h1>
    </div>
    <div
      class="add-page-grand-wrapper"
    >
      <div
        class="find-book-to-add"
      >
        <div
          class="top-block"
        >
          <h2>
            Enter the ISBN or ASIN code here
          </h2>
          <input
            type="text"
          />
        </div>
        <p>
          Click below to find the book you want to add. On the book's details page you'll be able to "Add this book to give away"
        </p>
        <a
          href="/searchresults//true"
          style="text-decoration: none;"
        >
          <button
            class="find-add-button"
          >
            <p>
              Find my
              <br />
              book to add
            </p>
          </button>
        </a>
      </div>
      <div
        class="add-page-footer"
      >
        <a
          href="/profile"
          style="text-decoration: none;"
        >
          <div
            class="user-button-wrapper"
          >
            <button
              class="user-button"
            />
            <p>
              User Home
            </p>
          </div>
        </a>
        <p>
          for books not yet in the catalogue, you'll have to enter the details manually on desktop
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
